frame/support/src/traits.rs

```diff
@@ -2228,6 +2228,41 @@ pub trait GetPalletVersion {
 	fn storage_version() -> Option<PalletVersion>;
 }
 
+/// A trait to check the position of inherent in a block.
+///
+/// This is typically implemented on runtime, through `construct_runtime!`.
+pub trait InherentPositionCheck<Block> {
+	/// Check the position of inherents in a block.
```

Comment:
While I can read this in the PR title that the goal is to bring inherents first, I find it strange that the trait is not defining this. I see two sensible paths: 

1. We think that any FRAME based runtime should have the inherents first, thus the trait should be called `EnsureInherentsAreFirst` or something like that, and the runitme implements this to ensures it. 
2. We think that any runtime can impose some ordering rule on the inherents, in which case the sentence IMO should say something like: ` Check the position of inherents in a block based on some arbitrary rule set.`

Reply:
Your naming leans toward 2, but I think given that you can't customize this in construct_runtime, 1 is more sensible.

Reply:
yes, I renamed.